lib/cartopy/mpl/gridliner.py

```diff
@@ -29,7 +29,7 @@
 from cartopy.crs import Projection, _RectangularProjection
 
 
-degree_locator = mticker.MaxNLocator(nbins=9, steps=[1, 2, 3, 6, 15, 18])
+degree_locator = mticker.MaxNLocator(nbins=9, steps=[1, 1.5, 1.8, 2, 3, 6, 10])
```

Comment:
i am intrigued what these values do and how changing them is better for mpl2 but no worse for mpl1.5

this change seems to have resulted in a number of image changes.

please could you talk me through the logic for this change?